src/NHibernate/Context/WcfOperationSessionContext.cs

```diff
@@ -50,3 +56,36 @@ public class WcfStateExtension : IExtension<OperationContext>
 		public void Detach(OperationContext owner) { }
 	}
 }
+#else
+// 6.0 TODO: remove the whole #else
+using System;
+using System.Collections;
+using NHibernate.Engine;
+
+namespace NHibernate.Context
+{
+	/// <summary>
+	/// Obsolete class not usable in the .Net Core and .Net Standard distributions of NHibernate. Use the
+	/// .Net Framework distribution of NHibernate if you need it. See
+	/// https://github.com/nhibernate/nhibernate-core/issues/1842
+	/// </summary>
+	[Obsolete("Not supported in the .Net Core and .Net Standard distributions of NHibernate", true)]
+	public class WcfOperationSessionContext : MapBasedSessionContext
+	{
+		public WcfOperationSessionContext(ISessionFactoryImplementor factory) : base(factory)
+		{
+			throw new NotSupportedException(
+				"WcfOperationSessionContext is currently supported only by the .Net Framework distribution of NHibernate");
```

Comment:
Need to change wording to emphasise that this is not supported in the current framework. The fact that it is only supported in full .NET Framework is not relevant.

Reply:
The thing is, current framework can be .Net Framework, while using the .Net Standard binaries. In such case stating it is not supported by the current framework while the lack of support actually comes from the binaries would be confusing.
Granted, that is a corner case. Normally by using NuGet no-one should ends up using the .Net Standard binaries under .Net Framework, and currently NHibernate .Net Standard binaries are not provided anywhere else than on NuGet.